DxOper.pas

```diff
@@ -141,7 +144,7 @@ function TDxOperator.IsMyCall: TCallCheckResult;
 
       D := M[x-1,y-1];
       //'?' matches any char
-      if not (C[x] in [C0[y], '?']) then Inc(D, W_D);
+      if not CharInSet(C[x], [C0[y], '?']) then Inc(D, W_D); //grg1

```

Comment:
Gerd,

When Delphi switched from single byte to double byte strings, its developers caused a lot of confusion by changing the meaning 
of "string" and "char". Char in D5 is single byte char, and Char in XE10 is a double byte char. This silent change of string 
types caused tons of problems. My own approach to updating old code for using in the Unicode versions of Delphi is as follows. 
Search all files in the project (Ctrl-Shift-F) for "string". Everywhere where a variable of type string is declared, figure out 
if it may contain non-ansi characters or not. If it is only ansi, change declaration to AnsiString, otherwise change it to 
UnicodeString. Ideally no vars of type string should be left in the program. Then I do the same for "char" and change the char 
types to either AnsiChar or WideChar. After these changes are performed, very little code needs manual inspection and correction.

73 Alex VE3NEA



On 2015-11-28 13:42, Gerd - DF9TS wrote:
> In DxOper.pas <https://github.com/VE3NEA/MorseRunner/pull/2#discussion_r46083013>:
>
>> @@ -141,7 +144,7 @@ function TDxOperator.IsMyCall: TCallCheckResult;
>>
>>        D := M[x-1,y-1];
>>        //'?' matches any char
>> -      if not (C[x] in [C0[y], '?']) then Inc(D, W_D);
>> +      if not CharInSet(C[x], [C0[y], '?']) then Inc(D, W_D); //grg1
>
> Delphi 10 creates warning for "in", warning suggests usage of CharInSet instead.
> So I replaced "in" with CharInSet() in all places, not just here
>
> —
> Reply to this email directly or view it on GitHub <https://github.com/VE3NEA/MorseRunner/pull/2/files#r46083013>.
>